plugins/orchestrator-common/scripts/openapi.sh

```diff
@@ -1,17 +1,70 @@
 #!/bin/bash
 pwd
-set -ex
+set -e
 
-npx openapi-typescript ./src/openapi/openapi.yaml -o ./src/auto-generated/api/models/schema.ts
-npx openapi-generator-cli generate -g asciidoc -i ./src/openapi/openapi.yaml -o ./src/auto-generated/docs/index.adoc
-npx yaml2json -f ./src/openapi/openapi.yaml
+OPENAPI_SPEC_FILE="./src/openapi/openapi.yaml"
+SCHEMA_FILE="./src/auto-generated/api/models/schema.ts"
+DEFINITION_FILE="./src/auto-generated/api/definition.ts"
+METADATA_FILE="./src/auto-generated/.METADATA.sha1"
 
-export FILE=./src/auto-generated/api/definition.ts
-echo '// GENERATED FILE. DO NOT EDIT.' > ${FILE}
-echo 'const OPENAPI = `' >> ${FILE}
-cat ./src/openapi/openapi.json >> ${FILE}
-echo '`' >> ${FILE}
-echo "export const openApiDocument = JSON.parse(OPENAPI);" >> ${FILE}
 
-rm ./src/openapi/openapi.json
-yarn openapi:prettier:fix
\ No newline at end of file
+openapi_generate() {
+    npx openapi-typescript ${OPENAPI_SPEC_FILE} -o ${SCHEMA_FILE}
+    npx openapi-generator-cli generate -g asciidoc -i ./src/openapi/openapi.yaml -o ./src/auto-generated/docs/index.adoc
+    npx yaml2json -f ${OPENAPI_SPEC_FILE}
+
+    echo '// GENERATED FILE. DO NOT EDIT.' > ${DEFINITION_FILE}
+    echo 'const OPENAPI = `' >> ${DEFINITION_FILE}
+    cat ./src/openapi/openapi.json >> ${DEFINITION_FILE}
+    echo '`' >> ${DEFINITION_FILE}
+    echo "export const openApiDocument = JSON.parse(OPENAPI);" >> ${DEFINITION_FILE}
+
+    rm ./src/openapi/openapi.json
+    yarn openapi:prettier:fix
+    NEW_SHA=$(openapi_checksum)
+    openapi_update ${NEW_SHA}
+}
+
+openapi_checksum(){
+    cat ${DEFINITION_FILE} ${SCHEMA_FILE} ${OPENAPI_SPEC_FILE} | sha1sum | awk '{print $1}' | tr -d '[:space:]'
+}
+
+openapi_update() {
+   echo "${1}" > "${METADATA_FILE}"
+}
+
+# Function to check if OpenAPI files are up-to-date
+openapi_check() {
+
+    if [ ! -f "${METADATA_FILE}" ]; then
+        echo "Error: Metadata file '${METADATA_FILE}' not found. Run 'generate_openapi.sh' first."
```

Comment:
Is this what you meant?
```suggestion
        echo "Error: Metadata file '${METADATA_FILE}' not found. Run 'yarn openapi:generate' first."
```

Reply:
thanks!